### PR TITLE
Apply requested changes on application form

### LINKED ISSUE #629: [Application] Confirm with STMAS that application form is ready
Once we have a first idea how the application form looks like (and the ping-pong with the organization is implemented in a first draft version), ask the STMAS for changes and feedback.

Blocked by #628 and #627 .

### PR DESCRIPTION
**Pull request by michael-markl**
Closes #629.

I've mostly made some fields optional, adjusted labels, and did some minor refactors.

### CODE REVIEW

File: administration/src/application/components/primitive-inputs/FileInputForm.tsx [hunk 1]
```diff
@@ -48,6 +50,75 @@ type State = {
 type ValidatedInput = AttachmentInput
 type Options = {}
 type AdditionalProps = {}
+
+const Component = <I,>({
+  state,
+  setState,
+  required,
+  validate,
+}: FormComponentProps<State, AdditionalProps, Options> & {
```
Comment (michael-markl) [inline administration/src/application/components/primitive-inputs/FileInputForm.tsx:59]: This was just extracted from below with the additional props "required" and "validate".

File: administration/src/application/components/forms/WorkAtDepartmentEntitlementForm.tsx [hunk 2]
```diff
@@ -29,21 +29,23 @@ const WorkAtDepartmentEntitlementForm: Form<State, Options, ValidatedInput, Addi
...
+      <SubForms.responsibility.Component
         label='Funktion oder Tätigkeit'
         state={state.responsibility}
         setState={useUpdateStateCallback(setState, 'responsibility')}
       />
       <h4>Tätigkeitsnachweis</h4>
       <p>
-        Hängen Sie hier bitte einen eingescannten oder abfotografierten Tätigkeitsnachweis an. Die Datei darf maximal{' '}
-        {FILE_SIZE_LIMIT_MEGA_BYTES} MB groß sein und muss im JPG, PNG oder PDF Format sein.
+        Hängen Sie hier bitte einen eingescannten oder abfotografierten Tätigkeitsnachweis an. {FileRequirementsText}
```
Comment (sarahsporck) [inline administration/src/application/components/forms/WorkAtDepartmentEntitlementForm.tsx:43]: Should this text not be adjusted to 
```suggestion
        Falls vorhanden: hängen Sie hier bitte einen eingescannten oder abfotografierten Tätigkeitsnachweis an. {FileRequirementsText}
```

File: administration/src/application/components/forms/VolunteerServiceEntitlementForm.tsx [hunk 3]
```diff
@@ -34,8 +34,7 @@ const VolunteerServiceEntitlementForm: Form<State, Options, ValidatedInput, Addi
       />
       <h4>Tätigkeitsnachweis</h4>
       <p>
-        Hängen Sie hier bitte einen eingescannten oder abfotografierten Tätigkeitsnachweis an. Die Datei darf maximal{' '}
-        {FILE_SIZE_LIMIT_MEGA_BYTES} MB groß sein und muss im JPG, PNG oder PDF Format sein.
+        Hängen Sie hier bitte einen eingescannten oder abfotografierten Tätigkeitsnachweis an. {FileRequirementsText}
```
Comment (sarahsporck) [inline administration/src/application/components/forms/VolunteerServiceEntitlementForm.tsx:37]: Shouldn't it be `Falls vorhanden: hängen Sie`_

**Review by sarahsporck — APPROVED** [review]

File: administration/src/application/components/forms/WorkAtDepartmentEntitlementForm.tsx (hunk 2 above)
Comment (michael-markl) [reply administration/src/application/components/forms/WorkAtDepartmentEntitlementForm.tsx:43]: Here, I agree, it seems I forgot to change it here. (And also make the certificate optional)

In general, I think we should only make the file upload optional for the cases where we contact the corresponding organization via email (#627) - this should be the case whenever the user can specify an organization & corresponding contact person.

File: administration/src/application/components/forms/VolunteerServiceEntitlementForm.tsx (hunk 3 above)
Comment (michael-markl) [reply administration/src/application/components/forms/VolunteerServiceEntitlementForm.tsx:37]: Here, we don't have the user specify an organization, so we should probably keep the certificate upload required. (see comment above/below)

**Pull request merged**